src/client/datascience/jupyter/kernels/kernelDependencyService.ts

```diff
@@ -80,6 +80,21 @@ export class KernelDependencyService implements IKernelDependencyService {
     public areDependenciesInstalled(interpreter: PythonEnvironment, _token?: CancellationToken): Promise<boolean> {
         return this.installer.isInstalled(Product.ipykernel, interpreter).then((installed) => installed === true);
     }
+
+    // The requirement for debugging is ipykernel v6 or newer
+    public areDebuggingDependenciesInstalled(
+        interpreter: PythonEnvironment,
+        _token?: CancellationToken
+    ): Promise<boolean> {
+        return this.installer.getVersion(Product.ipykernel, interpreter).then((version) => {
```

Comment:
Would async/await be more readable here? You forgot a 'catch' clause I think. Might be easier with try, await getVersion, parse versoin, catch - return false

Reply:
cleanup commit
although, I will check if we can accomplish this with the python api.

There's the issue that `python -m ipykernel --version` returns the ipython version instead of the ipykernel version. Hopefully the python api deals with that.